lib/uring/uring_stubs.c

```diff
@@ -184,6 +185,40 @@ ocaml_uring_submit_nop(value v_uring, value v_id) {
   CAMLreturn(Val_true);
 }
 
+value ocaml_uring_submit_timeout(value v_uring, value v_id, value v_clock, value v_timeout)
+{
+  CAMLparam4(v_uring, v_id, v_clock, v_timeout);
+  struct __kernel_timespec t;
```

Comment:
You can't pass a stack-allocated timeout. It needs to remain alive until submit time. @haesbaert is working on a PR to make this easier, I think.

Reply:
Ack, I have a silly allocator that can be used for this., it need a bit more work as I want to maybe rely on Uring.submit being called more promiscuously, which would simplify things a lot.

Reply:
Yes: allocate some memory, store a reference to it somewhere so it can't get GC'd, pass a pointer to it to Linux, then drop the reference at some point after submitting (or after completing, which is what the iovec stuff currently does).